### PR TITLE
Validate that names are not empty or duplicates

### PR DESCRIPTION
**Pull request by jasonharrison**

### CODE REVIEW

File: src/NameFieldSet.tsx [hunk 1]
```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
     this.setState({ names: [...this.state.names, ""] });
   }
 
+  private arrayContainsDuplicates = (a: string[]) => {
```
Comment (vdfdev) [inline src/NameFieldSet.tsx]: change variable name to a more descriptive name. Something like names or nameArray

File: src/NameFieldSet.tsx [hunk 2]
```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
...
+    const uniqueSet:Set<string> = new Set();
+    for(let i = 0; i <= a.length; i++) {
+      if (a[i] == null) {
```
Comment (vdfdev) [inline src/NameFieldSet.tsx]: why would we have an name as null? is this even possible?

File: src/NameFieldSet.tsx [hunk 3]
```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
...
+        continue
+      }
+      const value = a[i].toLowerCase();
```
Comment (vdfdev) [inline src/NameFieldSet.tsx]: value is not a descriptive name. replace it to "name" or something like this

File: src/NameFieldSet.tsx [hunk 4]
```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
     this.setState({ names: [...this.state.names, ""] });
   }
 
+  private arrayContainsDuplicates = (a: string[]) => {
+    const uniqueSet:Set<string> = new Set();
```
Comment (vdfdev) [inline src/NameFieldSet.tsx]: uniqueSet is too generic, replace with "nameSet" or something like this

File: src/NameFieldSet.tsx [hunk 5]
```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
...
+      uniqueSet.add(value)
+    }
+    return false;
+  }
+
+  private arrayContainsString = (a: string[], s: string) => {
+    return a.indexOf(s) !== -1;
+  }
+
+  private isValid = () => {
```
Comment (vdfdev) [inline src/NameFieldSet.tsx:81]: nice job with this function, very clear 👍

File: src/NameFieldSet.tsx [hunk 6]
```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
...
+      const value = a[i].toLowerCase();
+      if (uniqueSet.has(value)) {
+        return true;
+      }
+      uniqueSet.add(value)
+    }
+    return false;
+  }
+
+  private arrayContainsString = (a: string[], s: string) => {
```
Comment (vdfdev) [inline src/NameFieldSet.tsx]: a and s are not descriptive names. Replace with nameArray and name, respectively, for instance.

File: src/NameFieldSet.tsx (hunk 2 above)
Comment (jasonharrison) [reply src/NameFieldSet.tsx]: `== null` _also_ catches undefined (which is possible): https://stackoverflow.com/questions/2647867/how-to-determine-if-variable-is-undefined-or-null

**Review by vdfdev — APPROVED** [review]

**Pull request merged**